### PR TITLE
Add mTLS support into the example app

### PR DESCRIPTION
**Pull request by TheSpiritXIII**
Example usage:
```bash
go run ./examples/instrumentation/go-synthetic/ --tls-create-self-signed=true
curl -k https://localhost:8080/metrics
```

This PR builds on top of #569.

### CODE REVIEW

File: examples/instrumentation/go-synthetic/auth.go [hunk 1]
```diff
@@ -22,6 +41,170 @@ func isFlagSet(name string) bool {
...
+			return nil, fmt.Errorf("unable to create self-signed certificate: %w", err)
+		}
+		certPem := pem.EncodeToMemory(&pem.Block{
+			Type:  "CERTIFICATE",
+			Bytes: certBytes,
+		})
+
+		privateKeyBytes, err := x509.MarshalPKCS8PrivateKey(privateKey)
+		if err != nil {
+			log.Println("Unable to marshal private key", err)
```
Comment (bwplotka) [inline examples/instrumentation/go-synthetic/auth.go:181]: That's a bit odd, let's return error with wrap?

Comment (bwplotka) [reply examples/instrumentation/go-synthetic/auth.go:181]: same below

**Review by bwplotka — CHANGES_REQUESTED** [review]
LGTM, thanks!

Just error handling is off.

File: examples/instrumentation/go-synthetic/main.go [hunk 2]
```diff
@@ -230,12 +230,24 @@ func main() {
...
-			Addr:    *addr,
-			Handler: mux,
+			Addr:      *addr,
+			Handler:   mux,
+			TLSConfig: tlsConfig,
 		}
 
 		g.Add(func() error {
+			if tlsConfig != nil {
+				fmt.Printf("Starting server on %q with TLS\n", *addr)
```
Comment (macxamin) [inline examples/instrumentation/go-synthetic/main.go:247]: nit log.Println seems to be used in this file. Should we keep the standard?

File: examples/instrumentation/go-synthetic/main.go [hunk 3]
```diff
@@ -230,12 +230,24 @@ func main() {
...
+				return server.ListenAndServeTLS("", "")
+			}
+			fmt.Printf("Starting server on %q\n", *addr)
```
Comment (macxamin) [inline examples/instrumentation/go-synthetic/main.go:250]: nit same as above

**Review by macxamin — COMMENTED** [review]
LGTM

**Review by macxamin — APPROVED** [review]
LGTM

**Review by bwplotka — APPROVED** [review]
Thanks!

**Pull request merged**